__tests__/tooling/index.test.js

```diff
@@ -42,6 +42,138 @@ describe('#url()', () => {
   it('should add https:// if url does not start with a protocol', () => {
     expect(new Oas({ servers: [{ url: 'example.com' }] }).url()).toBe('https://example.com');
   });
+
+  it('should accept an index for servers selection', () => {
+    expect(new Oas({ servers: [{ url: 'example.com' }, { url: 'https://api.example.com' }] }).url(1)).toBe(
+      'https://api.example.com'
+    );
+  });
+
+  it('should default to first if selected is not valid', () => {
+    expect(new Oas({ servers: [{ url: 'https://example.com' }] }).url(10)).toBe('https://example.com');
+  });
+});
+
+describe('#splitUrl()', () => {
+  it('should split url into chunks', () => {
+    expect(
+      new Oas({
+        servers: [{ url: 'https://example.com/{path}' }],
+      }).splitUrl()
+    ).toStrictEqual([
+      { key: 'https://example.com/-0', type: 'text', value: 'https://example.com/' },
+      { key: 'path-1', type: 'variable', value: 'path', description: undefined, enum: undefined },
+    ]);
+  });
+
+  // Taken from here: https://github.com/readmeio/readme/blob/09ab5aab1836ec1b63d513d902152aa7cfac6e4d/packages/explorer/__tests__/PathUrl.test.jsx#L99-L111
+  it('should work for multiple path params', () => {
+    expect(
+      new Oas({
+        servers: [{ url: 'https://example.com/{a}/{b}/c' }],
+      }).splitUrl()
+    ).toHaveLength(5);
+    expect(
+      new Oas({
+        servers: [{ url: 'https://example.com/v1/flight/{FlightID}/sitezonetargeting/{SiteZoneTargetingID}' }],
+      }).splitUrl()
+    ).toHaveLength(4);
+  });
+
+  it('should create unique keys for duplicate values', () => {
+    expect(
+      new Oas({
+        servers: [{ url: 'https://example.com/{test}/{test}' }],
+      }).splitUrl()
+    ).toStrictEqual([
+      { key: 'https://example.com/-0', type: 'text', value: 'https://example.com/' },
+      { key: 'test-1', type: 'variable', value: 'test', description: undefined, enum: undefined },
+      { key: '/-2', type: 'text', value: '/' },
+      { key: 'test-3', type: 'variable', value: 'test', description: undefined, enum: undefined },
+    ]);
+  });
+
+  it('should return with description', () => {
+    expect(
+      new Oas({
+        servers: [{ url: 'https://example.com/{path}', variables: { path: { description: 'path description' } } }],
+      }).splitUrl()[1].description
+    ).toStrictEqual('path description');
+  });
+
+  it('should return with enum values', () => {
+    expect(
+      new Oas({
+        servers: [{ url: 'https://example.com/{path}', variables: { path: { enum: ['v1', 'v2'] } } }],
+      }).splitUrl()[1].enum
+    ).toStrictEqual(['v1', 'v2']);
+  });
+});
+
+describe('#variables([selected])', () => {
+  it('should return with list of variables', () => {
+    const variables = { path: { description: 'path description' } };
```

Comment:
Can you add `type: int` to these variable examples so they look more real?

Reply:
I dont think they have types? https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#serverVariableObject Unless i'm misunderstanding here.

Reply:
Ah no you're right, I was looking at this on mobile and was thinking I was looking at path params, not path vars.